scania-digital-design-system/sdds
# Feat/dropdown with shadow dom

**Describe pull-request**  
Version 2 of the dropdown. This component has all the functionality for multiselect/filter in one component. Also adds a hidden input element outside of the shadowDOM to be able to use the dropdown within a form. This includes two new element: sdds-dropdown-v2 and sdds-dropdown-option-v2. First one is a wrapper element that holds the dropdown button and most of the logic. The second is child element that should represent all options in the dropdown. As oppose to previous versions of the component we are using the default slot on the sdds-dropdown-option to set a label. This is closer to the browser implementation of select/option.

## Note!
This component does not accept a "data" prop. This is due to recommendation for webcomponent to not pass arrays/objects as props/attributes. Instead the user would have to map over and render how many sdds-dropdown-options they my need.

**Solving issue**  
Fixes: AB#3174

**How to test**  
1. Go to storybook link below.
2. Check in Components -> Dropdown-v2
3. Test the controls and check the API.  

**Suggested test steps**
